xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -376,63 +376,59 @@ void CPVRChannelGroup::SortByChannelNumber(void)
 
 CPVRChannelPtr CPVRChannelGroup::GetByClient(int iUniqueChannelId, int iClientID) const
 {
+  CPVRChannelPtr retval;
   CSingleLock lock(m_critSection);
 
-  for (unsigned int ptr = 0; ptr < m_members.size(); ptr++)
+  for (std::vector<PVRChannelGroupMember>::const_iterator it = m_members.begin(); !retval && it != m_members.end(); ++it)
   {
-    PVRChannelGroupMember groupMember = m_members.at(ptr);
-    if (groupMember.channel->UniqueID() == iUniqueChannelId &&
-        groupMember.channel->ClientID() == iClientID)
-      return groupMember.channel;
+    if ((*it).channel->UniqueID() == iUniqueChannelId &&
+        (*it).channel->ClientID() == iClientID)
+      retval = (*it).channel;
```

Comment:
I'll add a comment if it makes you happy ;) but seriously, if you assume that a for loop only breaks when at the end of the iteration, then it's just sloppy reading imo. The last part of the for doesn't always go to the next item either, e.g. When items are removed, further down in the patch. We don't write comments for those either. It's a bad assumption imo if you assume that a for loop with an iterator in it is only used when iterating over the whole vector and the break condition always is `it != members.end()`